shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/controllers/ManagedProcessorController.java

```diff
@@ -0,0 +1,171 @@
+
+        List<EventSource> eventSources = new ArrayList<>();
+        eventSources.add(EventSourceFactory.buildResourceInformer(kubernetesClient, LabelsBuilder.V2_OPERATOR_NAME,
+                ManagedProcessor.COMPONENT_NAME,
+                CamelIntegration.class));
+
+        return eventSources;
+    }
+
+    @Override
+    public UpdateControl<ManagedProcessor> reconcile(ManagedProcessor managedProcessor, Context context) {
+        String managedProcessorName = managedProcessor.getMetadata().getName();
+        String managedProcessorNamespace = managedProcessor.getMetadata().getNamespace();
+
+        LOGGER.info("Reconciling ManagedProcessor: '{}' in namespace '{}'",
+                managedProcessorName,
+                managedProcessorNamespace);
+
+        ManagedProcessorStatus processorStatus = managedProcessor.getStatus();
+
+        if (!processorStatus.isReady() && isTimedOut(processorStatus)) {
+            // notifyManagerOfFailure
+            processorStatus.markConditionFalse(ConditionTypeConstants.READY);
+            return UpdateControl.updateStatus(managedProcessor);
+        }
+
+        String integrationName = String.format("integration-%s", managedProcessorName);
+        CamelIntegration camelIntegration = managedProcessorService.fetchOrCreateCamelIntegration(managedProcessor, integrationName);
+
+        if (!camelIntegration.isReady()) {
+            LOGGER.info("CamelIntegration for the ManagedProcessor '{}' in namespace '{}' is not ready",
+                    managedProcessorName,
+                    managedProcessorNamespace);
+            if (!processorStatus.isConditionTypeFalse(ConditionTypeConstants.READY)) {
+                processorStatus.markConditionFalse(ConditionTypeConstants.READY);
+            }
+            return UpdateControl.updateStatus(managedProcessor).rescheduleAfter(executorPollIntervalMilliseconds);
+        } else if (!processorStatus.isConditionTypeTrue(ManagedProcessorStatus.CAMEL_INTEGRATION_AVAILABLE)) {
+            LOGGER.info("CamelIntegration for the ManagedProcessor '{}' in namespace '{}' is ready",
+                    managedProcessorName,
+                    managedProcessorNamespace);
+            processorStatus.markConditionTrue(ManagedProcessorStatus.CAMEL_INTEGRATION_AVAILABLE);
+        }
+
+        LOGGER.info("Managed Processor: '{}' in namespace '{}' is ready",
+                managedProcessorName,
+                managedProcessorNamespace);
+
+        // Only issue a Status Update once.
+        // This is a work-around for non-deterministic Unit Tests.
+        // See https://issues.redhat.com/browse/MGDOBR-1002
+        if (!managedProcessor.getStatus().isReady()) {
+            metricsService.onOperationComplete(managedProcessor, MetricsOperation.CONTROLLER_RESOURCE_PROVISION);
+            processorStatus.markConditionTrue(ConditionTypeConstants.READY);
```

Comment:
@lucamolteni For `ManagedBridge` the decision was taken not to use the high-level `Ready` condition. Instead, readiness is determined by all the `Conditions` on the `ManagedBridge` being `true`.

Reply:
isn't it what `.getStatus().isReady()` does? 

```
    @JsonIgnore
    public final boolean isReady() {
        return conditions.stream().anyMatch(c -> ConditionTypeConstants.READY.equals(c.getType()) && ConditionStatus.True.equals(c.getStatus()));
    }
```

Reply:
No because, it's looking explicitly only for the Condition with type `Ready` to be `true`.  Currently we've agreed that all Conditions on the resource should be `true` for it to be considered "ready", so essentially the code here needs to be:

```
return conditions.stream().allMatch((c)-> c.getStatus().equals(ConditionStatus.True))
```